traversal/common/Modifiers.java

```diff
@@ -0,0 +1,146 @@
+import com.vaticle.typeql.lang.pattern.variable.UnboundVariable;
+import com.vaticle.typeql.lang.query.builder.Sortable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.vaticle.typedb.common.collection.Collections.list;
+import static com.vaticle.typedb.common.collection.Collections.map;
+import static com.vaticle.typedb.common.collection.Collections.set;
+import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
+
+public class Modifiers {
+
+    Filter filter;
+    Sorting sorting;
+
+    public Modifiers() {
+        filter = new Filter(set());
+        sorting = new Sorting(list(), map());
+    }
+
+    public Filter filter() {
+        return filter;
+    }
+
+    public Modifiers filter(Filter filter) {
+        this.filter = filter;
+        return this;
+    }
+
+    public Sorting sorting() {
+        return sorting;
+    }
+
+    public Modifiers sorting(Sorting sorting) {
+        this.sorting = sorting;
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Modifiers other = (Modifiers) o;
+        return Objects.equals(filter, other.filter) && Objects.equals(sorting, other.sorting);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(filter, sorting);
+    }
+
+    public static class Filter {
+
+        Set<Identifier.Variable.Retrievable> variables;
+
+        private Filter(Set<Identifier.Variable.Retrievable> variables) {
+            this.variables = variables;
+        }
+
+        public static Filter create(Set<Identifier.Variable.Retrievable> variables) {
+            return new Filter(variables);
+        }
+
+        public static Filter create(List<UnboundVariable> vars) {
+            Set<Identifier.Variable.Retrievable> variables = new HashSet<>();
+            iterate(vars).map(v -> Identifier.Variable.of(v.reference().asName())).forEachRemaining(variables::add);
+            return new Filter(variables);
+        }
+
+        public Set<Identifier.Variable.Retrievable> variables() {
+            return variables;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Filter other = (Filter) o;
+            return variables.equals(other.variables);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(variables);
+        }
+    }
+
+    public static class Sorting {
+
+        public static Sorting EMPTY = new Sorting(list(), map());
+
+        List<Identifier.Variable.Retrievable> variables;
+        Map<Identifier.Variable.Retrievable, Boolean> ascending;
```

Comment:
Because the `ASC/DESC` is a part of `SortedIterator`, and it feels odd domain-wise to have a Modifier depending on an `SortedIterator.Order` order...
but you're right, the boolean isn't correct. THUS, I started moving the `ASC/DESC` out of Sorted iterator. HOWEVER this leads to a massive change that I will submit in a separate PR :P